core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/RegionSet.java

```diff
@@ -84,7 +90,10 @@ public static RegionSet create(String value) {
      * @param regions A collection of regions.
      */
     public static RegionSet create(Collection<String> regions) {
-        return new RegionSet(regions);
+        Validate.notEmpty(regions, "regions");
+        return new RegionSet(
+            regions.stream().map(s -> Validate.notEmpty(s, "region").trim()).collect(Collectors.toList())
```

Comment:
The way you have trimming logic, `create("  ")` will throw, but `create(Arrays.asList("  ")`, will not throw, which seems not good. Your tests miss this, because collectionFailList has case for `Arrays.asList(" ", ""),` but not just `Arrays.asList(" "),`.

Overall, when I commented https://github.com/aws/aws-sdk-java-v2/pull/4437#discussion_r1334927535 about adding cases for this, I wasn't thinking you'll make a behavior change. I think here with trimming and notEmpty it's starting to get into "validation" territory that we were avoiding. I think failing on null String/Collection seems fine because you didn't give us a value, but if there is a value, just accepting it as-is is what I had imagined.

Reply:
Yes, fair. Though, I meant to use `notBlank`, which checks whitespace and not length. I think we should still validate the very base case of what is a valid string to pass. Checking for blankness should be fine. We're not validating the pattern or whether it's a valid region, simply validating that there is actual content.